### PR TITLE
Add fi.laamaa.m8c

### PR DESCRIPTION
**Pull request by danbee**
### Please confirm your submission meets all the criteria.

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/laamaa/m8c/issues/110
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: fi.laamaa.m8c.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<application>
+  <id type="desktop">fi.laamaa.m8c.desktop</id>
```
Comment (hfiguiere) [inline fi.laamaa.m8c.appdata.xml:3]:
```suggestion
  <id>fi.laamaa.m8c</id>
```

File: fi.laamaa.m8c.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  <summary>Client for Dirtywave M8 headless</summary>
+  <project_license>MIT</project_license>
+  <metadata_license>CC0-1.0</metadata_license>
+  <description>
+    <p>
+       M8c is a client for the Dirtywave M8 hardware tracker running in headless mode
+    </p>
+  </description>
+  <url type="homepage">https://github.com/laamaa/m8c</url>
+</application>
```
Comment (hfiguiere) [inline fi.laamaa.m8c.appdata.xml:14]: missing `releases`, `content_rating`, `screenshots`

File: fi.laamaa.m8c.desktop [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=M8C
+GenericName=Dirtywave M8 Client
+Exec=/usr/bin/m8c
```
Comment (hfiguiere) [inline fi.laamaa.m8c.desktop:4]: this is incorrect.

1. it should be just the executable name (not absolute path)
2. it's definitely not in `/usr`.

File: fi.laamaa.m8c.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        path: fi.laamaa.m8c.appdata.xml
+      - type: file
+        path: fi.laamaa.m8c.desktop
+      - type: file
+        path: fi.laamaa.m8c.png
+    post-install:
+      - install -D -m644 fi.laamaa.m8c.appdata.xml /app/share/appdata/fi.laamaa.m8c.appdata.xml
+      - install -D -m644 fi.laamaa.m8c.desktop /app/share/applications/fi.laamaa.m8c.desktop
+      - install -D -m644 fi.laamaa.m8c.png /app/share/icons/hicolor/512x512/apps/fi.laamaa.m8c.png
+      - desktop-file-edit --set-icon='fi.laamaa.m8c' /app/share/applications/fi.laamaa.m8c.desktop
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:26]:
```suggestion
```

it's already set properly

File: fi.laamaa.m8c.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      - name: libserialport
+        buildsystem: simple
+        build-commands:
+          - ./autogen.sh
+          - ./configure --prefix=/app
+          - make install
+        sources:
+          - type: git
+            url: git://sigrok.org/libserialport
+            branch: master
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:39]: NEVER a branch.

File: fi.laamaa.m8c.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    post-install:
+      - install -D -m644 fi.laamaa.m8c.appdata.xml /app/share/appdata/fi.laamaa.m8c.appdata.xml
+      - install -D -m644 fi.laamaa.m8c.desktop /app/share/applications/fi.laamaa.m8c.desktop
+      - install -D -m644 fi.laamaa.m8c.png /app/share/icons/hicolor/512x512/apps/fi.laamaa.m8c.png
+      - desktop-file-edit --set-icon='fi.laamaa.m8c' /app/share/applications/fi.laamaa.m8c.desktop
+    modules:
+      - shared-modules/SDL2/SDL2-no-libdecor.json
+
+      - name: libserialport
+        buildsystem: simple
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:31]: this should be `autotools` and you don't need any build commands

File: fi.laamaa.m8c.yaml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+modules:
+  - name: m8c
+    buildsystem: simple
+    build-commands:
+      - PREFIX=/app make install
+    sources:
+      - type: git
+        url: https://github.com/laamaa/m8c.git
+        branch: main
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:15]: NEVER a branch. you need a tag or a commit.

File: fi.laamaa.m8c.yaml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        build-commands:
+          - ./autogen.sh
+          - ./configure --prefix=/app
+          - make install
+        sources:
+          - type: git
+            url: git://sigrok.org/libserialport
+            branch: master
+
+finish-args:
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:41]: finish-args usually comes before `modules`.

File: fi.laamaa.m8c.yaml [hunk 9]
```diff
@@ -0,0 +1,45 @@
+app-id: fi.laamaa.m8c
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:1]: I see no mention anywhere of `laamaa.fi`. Is this domain controlled by the developers?

File: fi.laamaa.m8c.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<application>
```
Comment (hfiguiere) [inline fi.laamaa.m8c.appdata.xml:2]:
```suggestion
<component type="desktop-application">
```

And the closing tag should be changed accordingly.

**Comment by danbee** [comment]
@hfiguiere Thanks for the feedback. I'll get this fixed up!

File: fi.laamaa.m8c.yaml (hunk 9 above)
Comment (danbee) [reply fi.laamaa.m8c.yaml:1]: This is the developers website address according to their GitHub profile. I assumed this was the right thing to use here but please let me know if it's not.

File: fi.laamaa.m8c.appdata.xml (hunk 2 above)
Comment (danbee) [reply fi.laamaa.m8c.appdata.xml:14]: Apologies, I wasn't aware these were required.

File: fi.laamaa.m8c.yaml (hunk 9 above)
Comment (hfiguiere) [reply fi.laamaa.m8c.yaml:1]: it's not about the developers (as a person) but rather the project. Something based on github repository would be more appropriate here

See https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: fi.laamaa.m8c.yaml [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: m8c
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:10]: does it support wayland? If yes then it's `fallback-x11` on the preceding line. other remove the wayland second.

(and I don't mean Xwayland, which is x11)

File: fi.laamaa.m8c.yaml [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+      - type: file
+        path: fi.laamaa.m8c.appdata.xml
+      - type: file
+        path: fi.laamaa.m8c.desktop
+      - type: file
+        path: icons/fi.laamaa.m8c.512.png
+      - type: file
+        path: icons/fi.laamaa.m8c.256.png
+      - type: file
+        path: icons/fi.laamaa.m8c.128.png
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:31]: you only need ONE since 128x128 or bigger, since don't come from upstream.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
the icon, .desktop and appstream files should be submitted upstream.

**Comment by hfiguiere** [comment]
bot, build fi.laamaa.m8c

File: fi.laamaa.m8c.yaml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+      - install -D -m644 fi.laamaa.m8c.64.png /app/share/icons/hicolor/64x64/apps/fi.laamaa.m8c.png
+      - install -D -m644 fi.laamaa.m8c.32.png /app/share/icons/hicolor/32x32/apps/fi.laamaa.m8c.png
+    modules:
+      - shared-modules/SDL2/SDL2-no-libdecor.json
+
+      - name: libserialport
+        buildsystem: autotools
+        sources:
+          - type: git
+            url: git://sigrok.org/libserialport
```
Comment (hfiguiere) [inline fi.laamaa.m8c.yaml:51]: `git:` url are a problem and this is why the build fails.

**Pull request closed without merging**